storage/replica.go

```diff
@@ -862,14 +865,22 @@ func (r *Replica) proposeRaftCommand(ctx context.Context, ba roachpb.BatchReques
 	}
 	cmdID := ba.GetOrCreateCmdID(r.store.Clock().PhysicalNow())
 	idKey := makeCmdIDKey(cmdID)
-	r.Lock()
-	r.pendingCmds[idKey] = pendingCmd
-	r.Unlock()
+
 	var errChan <-chan error
-	if r.proposeRaftCommandFn != nil {
-		errChan = r.proposeRaftCommandFn(idKey, raftCmd)
+	if r.quiesced {
```

Comment:
The lock needs to be held while accessing `r.quiesced` (I'm surprised and worried that the race detector didn't find this).